### PR TITLE
[MIG][12.0] sale_margin_delivered: Migration to v12.0

### PR DESCRIPTION
**Pull request by sergio-teruel**
cc @Tecnativa TT20182
@carlosdauden @chienandalu review please

### CODE REVIEW

**Comment by pedrobaeza** [comment]
Please rebase

**Comment by pedrobaeza** [comment]
TT?

**Comment by sergio-teruel** [comment]
This module is ready for merge

File: sale_margin_delivered/i18n/es.po [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+#. module: sale_margin_delivered
+#: model:ir.ui.view,arch_db:sale_margin_delivered.view_order_form
+msgid "% Margin"
+msgstr "% Margen"
+
+#. module: sale_margin_delivered
+#: model:ir.ui.view,arch_db:sale_margin_delivered.view_order_form
+msgid "Cost Price dlvd."
+msgstr ""
```
Comment (pedrobaeza) [inline sale_margin_delivered/i18n/es.po]: Translate it

File: sale_margin_delivered/i18n/es.po [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+#. module: sale_margin_delivered
+#: model:ir.ui.view,arch_db:sale_margin_delivered.view_order_form
+msgid "Margin"
+msgstr "Margen"
+
+#. module: sale_margin_delivered
+#: model:ir.model.fields,field_description:sale_margin_delivered.field_sale_order_line_margin_delivered
+#: model:ir.model.fields,field_description:sale_margin_delivered.field_sale_report_margin_delivered
+msgid "Margin Delivered"
+msgstr "Margin Entregados"
```
Comment (pedrobaeza) [inline sale_margin_delivered/i18n/es.po]:
```suggestion
msgstr "Margen de lo entregado"
```

File: sale_margin_delivered/i18n/es.po [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+
+#. module: sale_margin_delivered
+#: model:ir.model.fields,field_description:sale_margin_delivered.field_sale_order_line_margin_delivered_percent
+msgid "Margin Delivered Percent"
+msgstr "Porcentage de Margen de Entregados"
```
Comment (pedrobaeza) [inline sale_margin_delivered/i18n/es.po]:
```suggestion
msgstr "Porcentage de margen de lo entregado"
```

File: sale_margin_delivered/i18n/es.po [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+
+#. module: sale_margin_delivered
+#: model:ir.ui.view,arch_db:sale_margin_delivered.view_order_form
+msgid "Margin dlvd."
+msgstr "Margin entrg."
```
Comment (pedrobaeza) [inline sale_margin_delivered/i18n/es.po]:
```suggestion
msgstr "Margen entr."
```

File: sale_margin_delivered/i18n/es.po [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+
+#. module: sale_margin_delivered
+#: model:ir.model.fields,field_description:sale_margin_delivered.field_sale_order_line_purchase_price_delivery
+#, fuzzy
+msgid "Purchase Price Delivered"
+msgstr "Margin Entregados"
```
Comment (pedrobaeza) [inline sale_margin_delivered/i18n/es.po]:
```suggestion
msgstr "Precio de compra de lo entregado"
```

File: sale_margin_delivered/i18n/es.po [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+#: model:ir.model,name:sale_margin_delivered.model_sale_order_line
+msgid "Sales Order Line"
+msgstr "Línea de pedido de ventas"
+
+#. module: sale_margin_delivered
+#: model:ir.model,name:sale_margin_delivered.model_sale_report
+msgid "Sales Orders Statistics"
+msgstr "Estadísticas pedidos de ventas"
+
+#~ msgid "Transfer"
```
Comment (pedrobaeza) [inline sale_margin_delivered/i18n/es.po]: To remove

File: sale_margin_delivered/readme/USAGE.rst [hunk 7]
```diff
@@ -0,0 +1,17 @@
+#. Place a new *Sale Order* and add a line with an stockable product. Set a
```
Comment (pedrobaeza) [inline sale_margin_delivered/readme/USAGE.rst]:
```suggestion
#. Go to *Sales > Orders > Quotations*.
#. Create a new *Sales Order* and add a line with an stockable product. Set a
```

File: sale_margin_delivered/readme/USAGE.rst [hunk 8]
```diff
@@ -0,0 +1,17 @@
...
+   quantity higher than one.
+#. Confirm the *Sale Order* and deliver just a partial amount of product in the
```
Comment (pedrobaeza) [inline sale_margin_delivered/readme/USAGE.rst]:
```suggestion
#. Confirm the *Sales Order* and deliver just a partial amount of product in the
```

File: sale_margin_delivered/readme/USAGE.rst [hunk 9]
```diff
@@ -0,0 +1,17 @@
...
+
+For example:
+
+In an order line with a product at a cost of 10 and a sell price of 25 we
+deliver 2 of 3 units. Then, the reported margins would be:
+
+`margin`: 45 (3 * 15)
+`margin_delivered`: 30 (2 * 15)
+
+Additionaly, you can check the margin and the margin percent in the sale order
```
Comment (pedrobaeza) [inline sale_margin_delivered/readme/USAGE.rst]:
```suggestion
Additionally, you can check the margin and the margin percent in the sales order
```

File: sale_margin_delivered/readme/CONTRIBUTORS.rst [hunk 10]
```diff
@@ -0,0 +1,2 @@
+* Sergio Teruel <sergio.teruel@tecnativa.com>
```
Comment (pedrobaeza) [inline sale_margin_delivered/readme/CONTRIBUTORS.rst]: Put Tecnativa URL as parent

**Review by pedrobaeza — CHANGES_REQUESTED** [review]

**Review by chienandalu — COMMENTED** [review]
Runbot fails on `sale_margin_security` dependency. Maybe you have to rebase your branch

**Comment by pedrobaeza** [comment]
It's rebased AFAIK, but runbot has taken an old branch.

**Comment by chienandalu** [comment]
Browsing the commit files with github doesn't seem to be there:

image

**Comment by sergio-teruel** [comment]
Changes done!!

**Comment by sergio-teruel** [comment]
All green!!

**Comment by pedrobaeza** [comment]
But you have lost my suggested changes... what are you doing with git? :scream:

**Review by pedrobaeza — APPROVED** [review]

**Pull request merged**